src/parse/ris_live.go

```diff
@@ -5,45 +5,266 @@ import (
 	"encoding/json"
 	"fmt"
 	"log"
+	"net/netip"
 	"os"
+	"time"
 
 	"github.com/gorilla/websocket"
 )
 
 // RIS Live websocket url
 const socketUrl = "ws://ris-live.ripe.net/v1/ws/"
 
-func main() {
+var done chan interface{}
+var interrupt chan os.Signal
 
-	// create websocket connection to ris live websocket
-	conn, _, err := websocket.DefaultDialer.Dial(socketUrl, nil)
-	if err != nil {
-		log.Fatal("Websocket connection error:", err)
-		os.Exit(1)
-	}
-	defer conn.Close()
+type RisMessageData struct {
+	Host   string `json:"host,omitempty"`
+	Prefix string `json:"prefix,omitempty"`
+}
+
+type RisMessage struct {
+	Type string          `json:"type"`
+	Data *RisMessageData `json:"data"`
+}
 
+func receiveHandler(msgChannel chan common.BGPMessage, conn *websocket.Conn) {
+	//keep reading in new message from connection
 	for {
-		//keep reading in new message from connection
+
+		//take in next msg from connection
 		_, message, err := conn.ReadMessage()
 		if err != nil {
 			log.Println("Websocket read error:", err)
 			return
 		}
 
-		bgpMsg, err := parseLiveMessage(message)
+		//parse message to data structure
+		bgpMsgs, err := parseLiveMessage(message)
 		if err != nil {
 			log.Println("Error parsing BGP message", err)
 		} else {
-			fmt.Printf("Parsed BGP Message: %+v\n", bgpMsg) //prints parsed BGP msg
+			fmt.Printf("Parsed BGP Message: %+v\n", bgpMsgs) //prints parsed BGP msg
+		}
+
+		//put bgp messages into channel
+		for _, msg := range bgpMsgs {
+			msgChannel <- msg
+		}
+
+	}
+}
+
+// connects to ris live, parsing messages, and putting messages into msgChannel for processor
+func ParseRisLiveData(msgChannel chan common.BGPMessage) {
+
+	fmt.Println("starting...")
+
+	// create websocket connection to ris live websocket
+	conn, _, err := websocket.DefaultDialer.Dial(socketUrl, nil)
+	if err != nil {
+		log.Fatal("Websocket connection error:", err)
+		return
+	}
+	defer conn.Close()
+
+	go receiveHandler(msgChannel, conn)
+
+	fmt.Println("made connection")
+
+	/* Subscribe */
+	subscription1 := RisMessage{"ris_subscribe", &RisMessageData{"", "0.0.0.0/0"}}
```

Comment:
discussed already but this will of course listen for all updates in all the IPv4 space; no action needed here, just a note that the next steps are to pin a more specific subscription and/or pipe that in via configuration